src/rubrix/client/datasets.py

```diff
@@ -802,44 +809,65 @@ def prepare_for_training(self) -> "datasets.Dataset":
 
 
         """
-        import datasets
+        if framework is None:
 
-        has_annotations = False
-        for rec in self._records:
-            if rec.annotation is not None:
-                has_annotations = True
-                break
-
-        if not has_annotations:
-            return datasets.Dataset.from_dict({})
-
-        class_tags = ["O"]
-        class_tags.extend(
-            [
-                f"{pre}-{label}"
-                for label in sorted(self.__all_labels__())
-                for pre in ["B", "I"]
-            ]
-        )
-        class_tags = datasets.ClassLabel(names=class_tags)
+            import datasets
 
-        def spans2iob(example):
-            r = TokenClassificationRecord(
-                text=example["text"],
-                tokens=example["tokens"],
-                annotation=self.__entities_to_tuple__(example["annotation"]),
+            has_annotations = False
+            for rec in self._records:
+                if rec.annotation is not None:
+                    has_annotations = True
+                    break
+
+            if not has_annotations:
+                return datasets.Dataset.from_dict({})
+
+            class_tags = ["O"]
+            class_tags.extend(
+                [
+                    f"{pre}-{label}"
+                    for label in sorted(self.__all_labels__())
+                    for pre in ["B", "I"]
+                ]
             )
-            return class_tags.str2int(r.spans2iob(r.annotation))
+            class_tags = datasets.ClassLabel(names=class_tags)
 
-        ds = (
-            self.to_datasets()
-            .filter(self.__only_annotations__)
-            .map(lambda example: {"ner_tags": spans2iob(example)})
-        )
-        new_features = ds.features.copy()
-        new_features["ner_tags"] = [class_tags]
+            def spans2iob(example):
+                r = TokenClassificationRecord(
+                    text=example["text"],
+                    tokens=example["tokens"],
+                    annotation=self.__entities_to_tuple__(example["annotation"]),
+                )
+                return class_tags.str2int(r.spans2iob(r.annotation))
+
+            ds = (
+                self.to_datasets()
+                .filter(self.__only_annotations__)
+                .map(lambda example: {"ner_tags": spans2iob(example)})
+            )
+            new_features = ds.features.copy()
+            new_features["ner_tags"] = [class_tags]
+
+            return ds.cast(new_features)
+
+        elif framework == "spacy":
+
+            nlp = spacy.blank("en")
```

Comment:
Is the language relevant for the DocBin generation? Using `en` always should be enough?

Reply:
The language cannot be left blank, it requires a language code. Leaving it blank results in an error.

Reply:
Sorry, what I try to say is what happens when the dataset contains text in other language than English. Do it work in that case or should provide a way to setup the language?